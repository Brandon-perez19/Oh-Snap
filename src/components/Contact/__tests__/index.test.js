import React from 'react';
import ContactForm from '..';
import{render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

describe('Contact is rendering', () => {
    it('renders', () => {
        render(<ContactForm></ContactForm>)
    });

    it('matches snapshot', () => {
        const {asFragment} = render(<ContactForm></ContactForm>)
        expect(asFragment()).toMatchSnapshot();
    })

    it('renders h1 tag', () => {
        const {getByTestId} = render(<ContactForm></ContactForm>);
        expect(getByTestId('h1tag')).toHaveTextContent('Contact Me')
    })

    it('renders button', () => {
        const {getByTestId} = render(<ContactForm></ContactForm>);
        expect(getByTestId('btntag')).toHaveTextContent('Submit');
    })
});
