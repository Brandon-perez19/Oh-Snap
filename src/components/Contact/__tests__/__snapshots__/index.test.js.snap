// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact is rendering matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
       Contact Me 
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
           Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
           Your Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        data-testid="btntag"
        type="submit"
      >
         Submit 
      </button>
    </form>
  </section>
</DocumentFragment>
`;
